framework/tlvf/AutoGenerated/src/CmduMessageTx.cpp

```diff
@@ -84,11 +86,23 @@ bool CmduMessageTx::finalize(bool swap_needed)
     if (!m_cmdu_header)
         return false; 
 
-    if (m_cmdu_header->message_type() == eMessageType::VENDOR_SPECIFIC_MESSAGE &&
-        getClassCount() > 1) {
-        tlvVendorSpecific *tlv = dynamic_cast<tlvVendorSpecific *>(getClass(0).get());
-        if (tlv != nullptr) {
-            tlv->length() = getMessageLength() - (kCmduHeaderLength + kTlvHeaderLength);
+    // Update vendor specific length which is seperated to 3 classes
+    for (auto class_it = m_class_vector.begin(); class_it != m_class_vector.end(); class_it++) {
+        // Type is the first byte in the tlv buffer
+        uint8_t tlv_type = *(*class_it)->getStartBuffPtr();
+        if (tlv_type == uint8_t(eTlvType::TLV_VENDOR_SPECIFIC)) {
+            uint16_t* tlv_length = reinterpret_cast<uint16_t *>
```

Comment:
I think it would be better to do a static down-cast to tlvVendorSpecific here, and use the length field from there.

Reply:
How to cast it? tlvVendorSpecific is a complex class, not a struct...
It is possible to create a new struct that has type and length and cast into it. But I don't think it worth it.

Reply:
Since the TLV has been created with `addClass<tlvVendorSpecific>`, we know that `*class_it` (which is of type `BaseClass *`) is in fact of type `tlvVendorSpecific *`. So we can do a downcast to tlvVendorSpecific.

Note that this is only for getting the length field, so it's absolutely a nitpick.